broker/broker.go

```diff
@@ -308,8 +318,21 @@ func (b *CdnServiceBroker) checkDomain(domain, orgGUID string) error {
 }
 
 func (b *CdnServiceBroker) getHeaders(options Options) []string {
+	headers := make([]string, len(options.Headers))
+	copy(headers, options.Headers)
+	// Preserve original 'Host' header if forwarding to CloudFoundry's routing table.
```

Comment:
A user might specify a custom origin in order to back a domain name by something other than a CloudFoundry-hosted app. For instance, they could specify their own origin in order to back a CDN broker with an S3 bucket. 

While we haven't discussed whether we as a team want this functionality, we retained it so that our changes could be contributed upstream.

I'll take a look at improving the comment.